src/OpcUaStackCore/Base/Address.cpp

```diff
@@ -118,44 +119,70 @@ namespace OpcUaStackCore
 	void
 	Address::getIpsFromHostname(const std::string& hostname, std::vector<std::string>& ipVec)
 	{
-		struct addrinfo* result;
-		struct addrinfo* res;
-
-		int32_t error = getaddrinfo(hostname.c_str(), NULL, NULL, &result);
-	    if (error != 0) {
-	    	Log(Error, "getaddrinfo call error");
-	        return;
-	    }
-
-	    for (res = result; res != NULL; res = res->ai_next) {
-	        if (res->ai_addr->sa_family != AF_INET) {
-	    	   continue;
-	        }
-	        if (res->ai_socktype != SOCK_STREAM) {
-	        	continue;
-	        }
-	       	struct sockaddr_in *addr = (struct sockaddr_in *)res->ai_addr;
-	        std::string ip = inet_ntoa(addr->sin_addr);
-	        ipVec.push_back(ip);
-	    }
-
-	    freeaddrinfo(result);
+		using namespace boost::asio;
+
+		try {
+			io_service ioService;
+			ip::tcp::resolver resolver(ioService);
+			ip::tcp::resolver::query q(hostname, "");
+
+			resolver.async_resolve(q,
+				[hostname, &ipVec](const boost::system::error_code &ec, ip::tcp::resolver::iterator it) {
+
+			   if (!ec) {
+				   ipVec.push_back(it->endpoint().address().to_string());
+			   } else {
+				   Log(Error, "failed resolve host")
+						   .parameter("Hostname", hostname)
+						   .parameter("Error", ec.message());
+			   }
+
+			});
+
+			ioService.run();
```

Comment:
It was my first attempt, to do synchronously. Unfortunately, the interface of the resolver has changed since 1.66. Synchronous variant before 1.66 uses iterator which marked as deprecated in 1.66. The asynchronous variant has stable API for boost versions that supported by the stack. 

Your example works only for boost 1.66 and higher. Our docker containers with ubuntu 16.04 use 1.65.

Reply:
If using the deprecated iterator is ok. I can return the old variant.

Reply:
You can leave it that way. But write a short comment.